### PR TITLE
private keys can act as public keys, no need to supply the public key separately

### PR DESCRIPTION
**Pull request by axos88**

### CODE REVIEW

File: src/cert.rs [hunk 1]
```diff
@@ -26,52 +26,33 @@ fn ec_group(nid: Nid) -> EcGroup {
...
+    pkey: &PKey<pkey::Private>,
     domains: &[&str],
 ) -> Result<X509Req> {
     //
     // the csr builder
     let mut req_bld = X509ReqBuilder::new().expect("X509ReqBuilder");
 
     // set public key in builder
-    req_bld.set_pubkey(&pkey_pub).expect("set_pubkey");
+    req_bld.set_pubkey(&pkey).expect("set_pubkey");
```
Comment (algesten) [inline src/cert.rs:55]: this is the most surprising part to me. `set_pubkey` with a private key.

**Review by algesten — APPROVED** [review]
Ah! Today I learn something I didn't know about OpenSSL's API. Thanks!

**Comment by algesten** [comment]
I'll release this update later today (after work). It is a breaking change and I need to just check dependencies.

File: src/cert.rs (hunk 1 above)
Comment (axos88) [reply src/cert.rs:55]: Well if you think about it, the private key has all the information to calculate the public key from it. Actually if you look at the rust-openssl API, it defines two traits, HasPublic and HasPrivate. When accepting a PKey it defines which subtrait it wants. Public implements HasPublic, while Private implements both. I really like the solution :)

**Comment by axos88** [comment]
Thanks!

**Pull request merged**